src/Nuvemshop/Provider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace SocialiteProviders\Nuvemshop;
+
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://tiendanube.github.io/api-documentation/authentication
+ */
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'NUVEMSHOP';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'read_content',
+        'write_content',
+        'read_products',
+        'write_products',
+        'read_customers',
+        'write_customers',
+        'read_orders',
+        'write_orders',
+        'read_coupons',
+        'write_coupons',
+        'write_scripts',
+        'write_shipping'
+    ];
+
+    /**
+     * The separating character for the requested scopes.
+     *
+     * @var string
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * Returns the configured user id that we're authenticating with
+     *
+     * @return string
+     */
+    private function getClientId()
+    {
+        return $this->getConfig('client_id');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            sprintf(
+                'https://www.tiendanube.com/apps/%s/authorize',
+                $this->getClientId()
+            ),
+            $state
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://www.tiendanube.com/apps/authorize/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $accessToken = explode('.', $token)[0];
+        $userId = explode('.', $token)[1];
+
+        $response = $this->getHttpClient()->get(
+            "https://api.nuvemshop.com.br/v1/$userId/store",
+            [
+                RequestOptions::HEADERS => [
+                    'Authentication' => 'bearer ' . $accessToken,
```

Comment:
```suggestion
                    'Authentication' => 'Bearer ' . $accessToken,
```

Reply:
It should be lowercase as the [documentation](https://tiendanube.github.io/api-documentation/authentication) suggests: "The entire bearer word should be lowercase. Using a different word will result in a 401: unauthorized - invalid access token error."

Reply:
Up to you. Nevertheless, I would probably expect any server to support both lowercase and propercase. As lowercase is pretty uncommon.
cf. `\Illuminate\Http\Concerns\InteractsWithInput::bearerToken`

Reply:
I made two requests using Insomnia, one with lowercase and another with uppercase and received the following:

Uppercase:
![image](https://github.com/user-attachments/assets/d6a568c3-e7f8-41e7-b810-9443c86288e6)

Lowercase:
![image](https://github.com/user-attachments/assets/f0352778-6cd4-43ab-aba7-1ad3d4563a9f)